code/gui/mainwindow.h

```diff
@@ -27,14 +29,23 @@ class MainWindow : public QMainWindow {
 
 public slots:
     void openActuatorSetup();
+    void openSimulator();
 
 private slots:
+    // Button click events
     void on_move_button_clicked();
+    void on_controller_button_clicked();
+    // Mouse events
+    void mousePressEvent(QMouseEvent *event);
 
 private:
     Ui::MainWindow *ui;
     ActuatorSetup *actuator_setup_window;
-    std::shared_ptr<Actuator> m_controller;
+    SimulatorWindow *simulator_window;
+    std::shared_ptr<Controller> m_controller;
+	std::shared_ptr<Actuator> m_actuator;
+	std::shared_ptr<Simulator> m_simulator;
+    Controller::Type m_controller_type;

```

Comment:
So the plan is to use only one `Controller` pointer that may hold objects of any of its derived classes, i.e. either `Actuator` or `Simulator` (for now). Use the strategy design pattern to add the ability to hotswap the type of controllers.